### PR TITLE
product/search: Switch to CUE config

### PR DESCRIPTION
**Pull request by danielpoe**
Product config still uses old (non cue) config  and is also not prefixed with commerce

### CODE REVIEW

File: product/module.go [hunk 1]
```diff
@@ -46,25 +45,24 @@ func (*Module) Depends() []dingo.Module {
...
+	product: {
+		view:  {
+			template: *"product/product" | !=""
+		}
+		priceIsGross: bool | *true
+		generateSlug: bool | *true
+		slugAttributeCode: string | *"urlSlug"
+		fakeservice: {
+			enabled: bool | *false
+			currency: *"€" | !=""
```
[the remark this answers is not in the document]
Comment (danielpoe) [reply product/module.go:61]: Thanks. I adjusted the injection.
The fakeservice is only used in tests and introduced recently - I guess its ok without deprication

File: product/module.go [hunk 2]
```diff
@@ -46,25 +45,24 @@ func (*Module) Depends() []dingo.Module {
...
+			template: *"product/product" | !=""
+		}
+		priceIsGross: bool | *true
+		generateSlug: bool | *true
+		slugAttributeCode: string | *"urlSlug"
+		fakeservice: {
+			enabled: bool | *false
+			currency: *"€" | !=""  
+		}
+		pagination: defaultPageSize: number | *commerce.pagination.defaultPageSize
```
[the remark this answers is not in the document]
Comment (danielpoe) [reply product/module.go:63]: Inside ProductSearchService - there you can override the defaultPageSize
(but it itself defaults to the commerce.pagination.defaultPageSize)

It was also still wrong in the injection

**Comment by danielpoe** [comment]
Will want to test the config migration on the demoshop also before merge.
(The deprication and config references seems to behave unexpected)

**Comment by danielpoe** [comment]
@carstendietrich can you please check again.
The test with demoshop (checking if the legacy config still works) was ok

**Comment by carstendietrich** [comment]
> @carstendietrich can you please check again.
> The test with demoshop (checking if the legacy config still works) was ok

@danielpoe will have a look this afternoon and fix smaller stuff if needed so we can get this merged today.

**Pull request merged**